app/src/main/java/org/mozilla/fenix/components/metrics/LeanplumMetricsService.kt

```diff
@@ -58,46 +67,61 @@ class LeanplumMetricsService(private val application: Application) : MetricsServ
+            val applicationSetLocale = LocaleManager.getCurrentLocale(application)
+            val currentLocale = when (applicationSetLocale != null) {
+                true -> applicationSetLocale.isO3Language
+                false -> Locale.getDefault().isO3Language
+            }
+            if (!isLeanplumEnabled(currentLocale)) {
+                Log.i(LOGTAG, "Leanplum is not available for this locale: $currentLocale")
+                return@launch
+            }
 
-        Leanplum.start(application, hashMapOf(
-            "default_browser" to MozillaProductDetector.getMozillaBrowserDefault(application).orEmpty(),
-            "fennec_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.FIREFOX.productName),
-            "focus_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.FOCUS.productName),
-            "klar_installed" to installedApps.contains(MozillaProductDetector.MozillaProducts.KLAR.productName),
-            "fxa_signed_in" to application.settings().fxaSignedIn,
-            "fxa_has_synced_items" to application.settings().fxaHasSyncedItems,
-            "search_widget_installed" to application.settings().searchWidgetInstalled,
-            "fenix" to true
-        ))
+            when (token.type) {
+                Token.Type.Production -> Leanplum.setAppIdForProductionMode(token.id, token.token)
+                Token.Type.Development -> Leanplum.setAppIdForDevelopmentMode(token.id, token.token)
+                Token.Type.Invalid -> {
+                    Log.i(LOGTAG, "Invalid or missing Leanplum token")
+                    return@launch
+                }
+            }
+
+            LeanplumActivityHelper.enableLifecycleCallbacks(application)
+
+            val installedApps = MozillaProductDetector.getInstalledMozillaProducts(application)
+
+            Leanplum.start(application, hashMapOf(
+                "default_browser"
+                        to MozillaProductDetector.getMozillaBrowserDefault(application).orEmpty(),
+                "fennec_installed" to
+                        installedApps.contains(MozillaProductDetector.MozillaProducts.FIREFOX.productName),
+                "focus_installed"
+                        to installedApps.contains(MozillaProductDetector.MozillaProducts.FOCUS.productName),
+                "klar_installed"
+                        to installedApps.contains(MozillaProductDetector.MozillaProducts.KLAR.productName),
+                "fxa_signed_in" to application.settings().fxaSignedIn,
+                "fxa_has_synced_items" to application.settings().fxaHasSyncedItems,
+                "search_widget_installed" to application.settings().searchWidgetInstalled,
+                "fenix" to true
+            ))
+
+            withContext(Main) {
+                LeanplumInternal.setCalledStart(true)
+                LeanplumInternal.setHasStarted(true)
+                // Verify that this is currently set to true also
```

Comment:
Ah, this was meant as a TODO before landing :). We should verify that we do in fact always set `startedInBackround` to true during start. Then we can set it here as well to make sure.

Reply:
@MarcLeclair I've verified that this is the case now. So, you can remove the comment. As we're passing the application context here `LeanplumActivityHelper.currentActivity` will always be null and therefore `startedInBackground` will be true.